folding-schemes/src/frontend/noname/utils.rs

```diff
@@ -0,0 +1,157 @@
+use crate::Error;
+use ark_ff::PrimeField;
+use ark_r1cs_std::fields::fp::FpVar;
+use ark_r1cs_std::R1CSVar;
+use ark_relations::r1cs::SynthesisError;
+use noname::inputs::{parse_inputs, JsonInputs};
+use num_bigint::BigUint;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+pub const NONAME_CIRCUIT_EXTERNAL_INPUTS: &str =
+    "fn main(pub ivc_input: [Field; 2], external_inputs: [Field; 2]) -> [Field; 2] {
+    let xx = external_inputs[0] + ivc_input[0];
+    let yy = external_inputs[1] * ivc_input[1];
+    assert_eq(yy, xx);
+    return [xx, yy];
+}";
+
+pub const NONAME_CIRCUIT_NO_EXTERNAL_INPUTS: &str =
+    "fn main(pub ivc_input: [Field; 2]) -> [Field; 2] {
+    let out = ivc_input[0] * ivc_input[1];
+    return [out, ivc_input[1]];
+}";
```

Comment:
Why are these defined as string slices instead of using macros directly? Just wondering.

Reply:
Since those two noname circuit are only used for tests, isn't it more readable this way? How would you see things? (note that they are located [here](https://github.com/privacy-scaling-explorations/sonobe/blob/d2db27d434dd70c5078cbd1738a9e8e6e0577c4d/folding-schemes/src/frontend/noname/mod.rs#L127) now)

Reply:
Let's leave it as is then. Is indeed more readable as you say.